OpenRA.Game/Game.cs

```diff
@@ -863,14 +863,36 @@ public static void AddSystemLine(string text)
 			AddSystemLine("Battlefield Control", text);
 		}
 
-		public static void AddSystemLine(string name, string text)
+		public static void AddSystemLine(string prefix, string text)
 		{
-			OrderManager.AddChatLine(name, systemMessageColor, text, systemMessageColor);
+			AddChatLine(prefix, systemMessageColor, text, systemMessageColor, ChatPool.System);
 		}
 
 		public static void AddChatLine(string name, Color nameColor, string text)
 		{
-			OrderManager.AddChatLine(name, nameColor, text, chatMessageColor);
+			AddChatLine(name, nameColor, text, chatMessageColor, ChatPool.Chat);
+		}
+
+		public static void AddChatLine(string name, Color nameColor, string text, Color textColor,  ChatPool pool)
+		{
+			OrderManager.AddChatLine(new ChatLine(name, nameColor, text, textColor, pool));
+		}
+
+		public static void AddTranscribedChatLine(string text)
+		{
+			if (Settings.Game.ChatPoolFilters.HasFlag(ChatPoolFilters.Transcriptions))
+				AddChatLine("Battlefield Control", systemMessageColor, text, systemMessageColor, ChatPool.Transcriptions);
+		}
+
+		public static void AddFeedbackChatLine(string text)
+		{
+			if (Settings.Game.ChatPoolFilters.HasFlag(ChatPoolFilters.Feedback))
+				AddChatLine("Battlefield Control", systemMessageColor, text, systemMessageColor, ChatPool.Feedback);
+		}
+
+		public static void AddMissionChatLine(string name, Color nameColor, string text)
+		{
+			AddChatLine(name, nameColor, text, chatMessageColor, ChatPool.Mission);
```

Comment:
One thing that bothers me with this implementation is that now there are a bunch of add-chat-line methods cluttering `Game.cs`. I thought about moving them to their own file but I couldn't figure out how to do it. IIRC it had something to do with passing the `OrderManager` instance. I didn't dig too deep into that though. And of course this can be done in a follow-up.